sql/mysql_db_structure.sql

```diff
@@ -364,23 +372,31 @@ CREATE TABLE `problem` (
   `problemtext` longblob DEFAULT NULL COMMENT 'Problem text in HTML/PDF/ASCII',
   `problemtext_type` varchar(4) DEFAULT NULL COMMENT 'File type of problem text',
   PRIMARY KEY  (`probid`),
-  KEY `externalid` (`externalid`(190))
+  KEY `externalid` (`externalid`(190)),
+  KEY `special_run` (`special_run`),
+  KEY `special_compare` (`special_compare`),
+  CONSTRAINT `problem_ibfk_1` FOREIGN KEY (`special_run`) REFERENCES `executable` (`execid`) ON DELETE SET NULL,
+  CONSTRAINT `problem_ibfk_2` FOREIGN KEY (`special_compare`) REFERENCES `executable` (`execid`) ON DELETE SET NULL
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci COMMENT='Problems the teams can submit solutions for';
 
 --
 -- Table structure for table `rankcache`
 --
 
 CREATE TABLE `rankcache` (
-  `cid` int(4) unsigned NOT NULL COMMENT 'Contest ID',
-  `teamid` int(4) unsigned NOT NULL COMMENT 'Team ID',
+  `cid` int(4) unsigned NOT NULL COMMENT 'Unique contest ID',
+  `teamid` int(4) unsigned NOT NULL COMMENT 'Unique team ID',
   `points_restricted` int(4) unsigned NOT NULL DEFAULT 0 COMMENT 'Total correctness points (restricted audience)',
   `totaltime_restricted` int(4) NOT NULL DEFAULT 0 COMMENT 'Total penalty time in minutes (restricted audience)',
   `points_public` int(4) unsigned NOT NULL DEFAULT 0 COMMENT 'Total correctness points (public)',
   `totaltime_public` int(4) NOT NULL DEFAULT 0 COMMENT 'Total penalty time in minutes (public)',
   PRIMARY KEY (`cid`,`teamid`),
   KEY `order_restricted` (`cid`,`points_restricted`,`totaltime_restricted`) USING BTREE,
-  KEY `order_public` (`cid`,`points_public`,`totaltime_public`) USING BTREE
+  KEY `order_public` (`cid`,`points_public`,`totaltime_public`) USING BTREE,
```

Comment:
Unrelated, but is there any reason we have `USING BTREE` here (and only for these two lines)?

Reply:
Yes! The fact that we can compare on this key is used here: https://github.com/DOMjudge/domjudge/blob/master/webapp/src/Service/ScoreboardService.php#L181

Reply:
But does that only work when we have btree explicitly set?

Reply:
No, but with BTREE that query is better optimized. Also BTREE is the default index type nowadays (instead of HASH), but still better to be explicit about it.